src/dynamo/models/comment.ts

```diff
@@ -111,86 +93,90 @@ const batchHydrate = async (items: DocumentClient.ItemList | undefined): Promise
 };
 
 const comment = {
-  getById: async (id: string): Promise<Comment | undefined> => hydrate((await client.get(id)).Item),
+  getById: async (id: string): Promise<Comment | undefined> =>
+    hydrate((await client.get(id)).Item as UnhydratedComment),
   queryByParentAndType: async (
     parent: string,
     lastKey?: DocumentClient.Key,
-  ): Promise<{ items?: DocumentClient.ItemList; lastKey?: DocumentClient.Key }> => {
+  ): Promise<{ items?: Comment[]; lastKey?: DocumentClient.Key }> => {
+    //Using keyof .. provides static checking that the attribute exists in the type. Also its own const b/c inline "as keyof" not validating
+    const parentAttr: keyof UnhydratedComment = 'parent';
+
     const result = await client.query({
       IndexName: 'ByParent',
       KeyConditionExpression: `#p1 = :parent`,
       ExpressionAttributeValues: {
         ':parent': parent,
       },
       ExpressionAttributeNames: {
-        '#p1': FIELDS.PARENT,
+        '#p1': parentAttr,
       },
       ExclusiveStartKey: lastKey,
       ScanIndexForward: false,
       Limit: 10,
     });
 
     return {
-      items: await batchHydrate(result.Items),
+      items: await batchHydrate(result.Items as UnhydratedComment[]),
       lastKey: result.LastEvaluatedKey,
     };
   },
   queryByOwner: async (
     owner: string,
     lastKey?: DocumentClient.Key,
-  ): Promise<{ items?: DocumentClient.ItemList; lastKey?: DocumentClient.Key }> => {
+  ): Promise<{ items?: Comment[]; lastKey?: DocumentClient.Key }> => {
+    const ownerAttr: keyof UnhydratedComment = 'owner';
```

Comment:
As an example of the static checking, if the string here was "ownar" then typescript would complain it isn't a known key of the UnhydratedComment type